### PR TITLE
Added Linear tool

### PR DESCRIPTION
**Pull request by dundermain**
This tool allows to perform the following basic operations using linear tool API

- Get issue details from the issue id
- Create an issue
- Update an issue with issue id
- Get all the issues assigned to a user 
- Get all the workflow issues associated with workflow id
- Get high-priority issues

This pull request completes #1331

### CODE REVIEW

File: cookbook/tools/linear_tools.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+agent = Agent(
+    name="Linear Tool Agent",
+    tools=[LinearTool()],
+    show_tool_calls=True,
+    markdown=True,
+)
+
+
+agent.print_response("Show all the assigned to user id '69069")
```
Comment (manthanguptaa) [inline cookbook/tools/linear_tools.py:12]: show all the assigned "issues" to user id

File: cookbook/tools/linear_tools.py
Comment (manthanguptaa) [inline cookbook/tools/linear_tools.py:1]: can you format all the ids to be taken from variables?

File: phi/tools/linear_tools.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+from phi.tools import Toolkit
+from phi.utils.log import logger
+
+
+class LinearTool(Toolkit):
+    def __init__(self):
+        super().__init__(name="linear tools")
+        self.api_token = getenv("LINEAR_API_KEY")
+        self.endpoint = "https://api.linear.app/graphql"
+        self.headers = {"Authorization": f"Bearer {self.api_token}"}
```
Comment (manthanguptaa) [inline phi/tools/linear_tools.py:12]: You haven't registered the functions. Take example from https://github.com/phidatahq/phidata/blob/main/phi/tools/slack.py

File: phi/tools/linear_tools.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Request error: {e}")
+            raise
+
+        except Exception as e:
+            logger.error(f"Unexpected error: {e}")
+            raise
+
+    # issue related methods
```
Comment (manthanguptaa) [inline phi/tools/linear_tools.py:37]: unnecessary comment

File: phi/tools/linear_tools.py
Comment (manthanguptaa) [inline phi/tools/linear_tools.py:1]: improve the doc strings for function by also adding function signature details

Comment (manthanguptaa) [inline phi/tools/linear_tools.py:1]: Add return type to all the functions

File: phi/tools/linear_tools.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
+import requests
+from os import getenv
+from phi.tools import Toolkit
+from phi.utils.log import logger
+
+
+class LinearTool(Toolkit):
+    def __init__(self):
+        super().__init__(name="linear tools")
+        self.api_token = getenv("LINEAR_API_KEY")
```
Comment (manthanguptaa) [inline phi/tools/linear_tools.py:10]: Add if condition to check if the key exists. If it doesn't exists then raise an error

File: phi/tools/linear_tools.py
Comment (manthanguptaa) [inline phi/tools/linear_tools.py:1]: The formatting also looks off in the file. Try running `./scripts/format.sh`

Comment (dundermain) [reply phi/tools/linear_tools.py:1]: Hey @manthanguptaa, yeah, I had run validation, but I think some of the formatting was not changed by Ruff. So, this time, I have cross-checked and changed the formatting manually.

**Comment by manthanguptaa** [comment]
@dundermain I tried running the tool but it's giving me 400 bad request. 
Screenshot 2024-11-12 at 4 35 30 PM

Also instead of using `|` use `Optional` or `Union`

**Comment by dundermain** [comment]
Hey @manthanguptaa, I am currently looking into the issue. There seems to be an issue with the type definition in GraphQL. I will experiment with it, and if it works, I will raise the PR. I have replaced "|" with "Optional". So once I debug the Linear issue, will raise the PR for both changes.

**Comment by dundermain** [comment]
@manthanguptaa  I have made the required changes. Please test it again and let me know if you face any issues. I tested it from my end, and it is working.

**Comment by manthanguptaa** [comment]
@dundermain allow me some time. I will most probably be reviewing this on the weekend

**Review by manthanguptaa — APPROVED** [review]
LGTM and works well. If you can also add how to get the different UUIDs on top of the cookbook it will be really helpful for the user.

**Comment by dundermain** [comment]
Thanks a lot, @manthanguptaa
I will work on the UUIDs, but I need some context. By UUIDs, do you mean the different IDs like team ID, state ID, and user ID in the linear tool?

**Review by ysolanky — APPROVED** [review]

**Pull request merged**